src/app/checkout/step-2/step-2.component.spec.js

```diff
@@ -308,6 +308,77 @@ describe('checkout', () => {
         expect(self.controller.paymentFormState).toBe('success')
         expect(self.controller.getContinueDisabled()).toBe(false)
       })
+
+      it('should return true when cvv validity is false and card type is credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = false
+        self.controller.handlePaymentChange({'card-type': 'visa'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(true)
+      })
+
+      it('should return true when cvv validity is true and card type is credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = true
+        self.controller.handlePaymentChange({'card-type': 'visa'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(false)
+      })
+
+      it('should return true when cvv validity is undefined and card type is credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = undefined
+        self.controller.handlePaymentChange({'card-type': 'visa'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(false)
+      })
+
+      it('should return false when cvv validity is false and card type is not credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = false
+        self.controller.handlePaymentChange({'account-type': 'checking'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(false)
+      })
+
+      it('should return false when cvv validity is undefined and card type is not credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = undefined
+        self.controller.handlePaymentChange({'account-type': 'checking'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(false)
+      })
+
+      it('should return false when cvv validity is true and card type is not credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = true
+        self.controller.handlePaymentChange({'account-type': 'checking'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(false)
+      })
```

Comment:
You could wrap all these in a description `card type is credit card` or `card type is not credit card`